ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -79,8 +80,13 @@ S3Connection::ListObjects(const std::string &prefix, std::vector<std::string> &o
     listSingle ? request.SetMaxKeys(1) : request.SetMaxKeys(batchSize);
 
     Aws::S3Crt::Model::ListObjectsV2Outcome outcomes = _s3CrtClient.ListObjectsV2(request);
-    if (!outcomes.IsSuccess())
-        return (1);
+    Aws::Http::HttpResponseCode resCode = outcomes.GetError().GetResponseCode();
```

Comment:
You should get rescode only in case there is a failure, i.e. inside the body of the if condition at line 84. Why extract a rescode if you wont need it in case of success. Also reducing the scope of this variable will prevent the bug in the lines 104-109, where rescode from the first request is used instead of updating resCode each time a request fails.

Reply:
Ah thankyou, that's a good idea, fixed now.